### PR TITLE
Add an error message class for a case where a trait is expected

### PR DESCRIPTION
**Pull request by maseev**
This PR is related to #1589. Specifically, it adds a new error message class for the following case - `Checking.scala:551`

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -1888,4 +1888,32 @@ object messages {
...
+
+  case class TraitIsExpected(tref: TypeRef)(implicit ctx: Context) extends Message(TraitIsExpectedID) {
+    val kind = "Syntax"
+    val msg = hl"$tref is not a trait"
+    val explanation = {
+      val errorCodeExample =
+        """class A
+          |class B
+          |
+          |val a = new A with B // will fail with a compile error - B is not a trait"""
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: You should `stripMargin` here

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 2]
```diff
@@ -1888,4 +1888,32 @@ object messages {
...
+      val codeExample =
+        """class A
+          |trait B
+          |
+          |val a = new A with B // compiles normally"""
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: You should `stripMargin` here

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 3]
```diff
@@ -1888,4 +1888,32 @@ object messages {
     }
     val explanation = "A statement is either an import, a definition or an expression."
   }
+
+  case class TraitIsExpected(tref: TypeRef)(implicit ctx: Context) extends Message(TraitIsExpectedID) {
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: I would rather pass the symbol instead of the `TypeRef`

**Comment by maseev** [comment]
@allanrenucci all done. PTAL.

**Comment by maseev** [comment]
It seems that the CLA check is still pending. Is there anything I can do to make it pass? :worried:

**Comment by nicolasstucki** [comment]
Sometimes the CLA stalls

**Pull request merged**